### PR TITLE
[terraform-aws] add terraform aws provisioning scripts

### PR DESCRIPTION
**Pull request by tmc**
Mirrors upstream kube-up.sh behavior.

### CODE REVIEW

File: terraform-aws/LICENSE [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Copyright (c) 2015, Travis Cline <travis.cline@gmail.com>
```
Comment (brendandburns) [inline terraform-aws/LICENSE]: I think we want everything under kubernetes to be covered by the Kubernetes apache license and copyright the kubernetes owners.

If that's not ok, this probably belongs in a different repo.

Comment (tmc) [inline terraform-aws/LICENSE]: That's totally ok.

**Comment by bgrant0607** [comment]
cc @mikedanese @justinsb

**Comment by foxish** [comment]
@bgrant0607 can this be closed?

**Comment by zmerlynn** [comment]
This doesn't look very current, and both [`kops`](https://github.com/kubernetes/kops) and [`kube-aws`](https://coreos.com/kubernetes/docs/latest/kubernetes-on-aws.html) can output Terraform and are maintained (and there are probably others).

**Comment by mikedanese** [comment]
If we want this we should accept it into github.com/kubernetes/kube-deploy anyway and not contrib.

**Pull request closed without merging**